tests/qos/test_qos_dscp_mapping.py

```diff
@@ -0,0 +1,301 @@
+        inner_dst_pkt_ip = DUMMY_INNER_DST_IP
+        inner_src_pkt_ip = DUMMY_INNER_SRC_IP
+        ptf_src_mac = ptfadapter.dataplane.get_mac(0, ptf_src_port_id)
+
+        pytest_assert(dut_qos_maps.get("dscp_to_tc_map") and dut_qos_maps.get("tc_to_queue_map"),
+                      "No QoS map found on DUT")
+
+        for rotating_dscp in range(0, 64):
+            if decap_mode == "uniform":
+                outer_dscp = rotating_dscp
+                inner_dscp = DEFAULT_DSCP
+            elif decap_mode == "pipe":
+                outer_dscp = DEFAULT_DSCP
+                inner_dscp = rotating_dscp
+
+            pkt, exp_pkt = create_ipip_packet(outer_src_mac=ptf_src_mac,
+                                              outer_dst_mac=router_mac,
+                                              outer_src_pkt_ip=outer_src_pkt_ip,
+                                              outer_dst_pkt_ip=outer_dst_pkt_ip,
+                                              outer_dscp=outer_dscp,
+                                              inner_src_pkt_ip=inner_src_pkt_ip,
+                                              inner_dst_pkt_ip=inner_dst_pkt_ip,
+                                              inner_dscp=inner_dscp,
+                                              decap_mode=decap_mode)
+
+            queue_val = get_dscp_to_queue_value(rotating_dscp, dut_qos_maps.get("dscp_to_tc_map").get("AZURE"),
+                                                dut_qos_maps.get("tc_to_queue_map").get("AZURE"))
+
+            global output_table
+
+            if queue_val is None:
+                logger.info("No queue found for dscp {} on DUT".format(inner_dscp))
+                output_table.append([rotating_dscp, "No queue found", "N/A"])
+                continue
+
+            # Clear queue counters
+            duthost.command("sonic-clear queuecounters")
+            time.sleep(2)
+
+            try:
+                dst_ptf_port_id = send_and_verify_traffic(ptfadapter=ptfadapter,
+                                                          pkt=pkt,
+                                                          exp_pkt=exp_pkt,
+                                                          ptf_src_port_id=ptf_src_port_id,
+                                                          ptf_dst_port_ids=ptf_dst_port_ids)
+
+            except Exception as e:
+                raise (e)
+
+            dut_egress_port = get_dut_pair_port_from_ptf_port(duthost, tbinfo, dst_ptf_port_id)
+            pytest_assert(dut_egress_port, "No egress port on DUT found for ptf port {}".format(dst_ptf_port_id))
+
+            egress_queue_count, _ = get_egress_queue_count(duthost, dut_egress_port, queue_val)
+            verification_success = abs(egress_queue_count - DEFAULT_PKT_COUNT) < TOLERANCE
+
+            if test_params["dut_asic_type"] == "broadcom":
+                pytest_assert(verification_success, "Received {} packets on queue {} instead of {}".format(
```

Comment:
Why do we need to check broadcom asic?

Reply:
I am only running the pytest assert on broadcom asic switches because other switches have unknown behavior, hence I am tabulating the results.

Reply:
Can you remove the ASIC change and make it generic?
If the egress queue is not expected, just record the error and put the result into the output_table.
At the end of testing, raise an error if any error happened.

Reply:
Ok